### PR TITLE
Added regex tests and more negative tests

### PR DESCRIPTION
**Pull request by ShreyasRmsft**

### CODE REVIEW

File: Agent.Plugins.Log.TestResultParser.Parser/Parser/Node/Jest/States/JestParserStateExpectingTestResults.cs [hunk 1]
```diff
@@ -18,8 +18,8 @@ public JestParserStateExpectingTestResults(ParserResetAndAttemptPublish parserRe
         {
             RegexsToMatch = new List<RegexActionPair>
             {
-                new RegexActionPair(JestRegexs.PassedTestCase, PassedTestCaseMatched),
                 new RegexActionPair(JestRegexs.FailedTestCase, FailedTestCaseMatched),
```
Comment (singhsarab) [inline Agent.Plugins.Log.TestResultParser.Parser/Parser/Node/Jest/States/JestParserStateExpectingTestResults.cs]: Why this change ?

File: Agent.Plugins.UnitTests/MochaTestResultParserTests/MochaTestResultParserRegexTests.cs [hunk 2]
```diff
@@ -34,7 +34,7 @@ public void RegexPatternTest(string regexPattern)
 
         public static IEnumerable<object[]> GetRegexPatters()
```
Comment (nigurr) [inline Agent.Plugins.UnitTests/MochaTestResultParserTests/MochaTestResultParserRegexTests.cs]: enable spell checker 🍡

Comment (ShreyasRmsft) [reply Agent.Plugins.UnitTests/MochaTestResultParserTests/MochaTestResultParserRegexTests.cs]: will fix
